go-controller/pkg/util/node_annotations_unit_test.go

```diff
@@ -0,0 +1,298 @@
+package util
+
+import (
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube"
+	ovntest "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/testing"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"net"
+	"reflect"
+	"testing"
+)
+import "fmt"
+
+func TestL3GatewayConfig_MarshalJSON(t *testing.T) {
+	vlanid := uint(1024)
+	tests := []struct {
+		desc       string
+		inpL3GwCfg *L3GatewayConfig
+		expOutput  []byte
+	}{
+		{
+			desc:       "test empty config i.e gateway mode disabled by default",
+			inpL3GwCfg: &L3GatewayConfig{},
+			expOutput:  []byte(`{"mode":""}`),
+		},
+		{
+			desc: "test gateway mode set to local and verify that node-port-enable is set to false by default",
+			inpL3GwCfg: &L3GatewayConfig{
+				Mode: config.GatewayModeLocal,
+			},
+			expOutput: []byte(`{"mode":"local","node-port-enable":"false"}`),
+		},
+		{
+			desc: "test VLANID not nil",
+			inpL3GwCfg: &L3GatewayConfig{
+				Mode:   config.GatewayModeLocal,
+				VLANID: &vlanid,
+			},
+			expOutput: []byte(`{"mode":"local","node-port-enable":"false","vlan-id":"1024"}`),
```

Comment:
This should actually be an error: `VLANID` is only allowed with shared gateway mode

Reply:
Thanks for catching this, however, I should be able to modify line 39 to be `Mode: config.GatewayModeLocal` and L42 to be `expOutput: []byte(`{"mode":"local","node-port-enable":"false","vlan-id":"1024"}`)` , right?

Reply:
Assuming that when you say `GatewayModeLocal` you mean `GatewayModeShared` and that when you say `"mode":"local"` you mean `"mode":"shared'`, then yes.

But also, I meant that this is pointing out a bug in the (non-test) code; it ought to be throwing an error in the case you have now.

Reply:
Thanks for the correction again, Below should be the corrected test case
```
desc: "test VLANID not nil",
			inpL3GwCfg: &L3GatewayConfig{
				Mode:   config.GatewayModeShared,
				VLANID: &vlanid,
			},
			expOutput: []byte(`{"mode":"shared","node-port-enable":"false","vlan-id":"1024"}`),
```
Let me open a Github issue to track that the non-test code needs to throw an error for the below scenario
```
inpL3GwCfg: &L3GatewayConfig{
				Mode:   config.GatewayModeLocal,
				VLANID: &vlanid,
			},
```

Reply:
FYI Opened issue https://github.com/ovn-org/ovn-kubernetes/issues/1478